build/Rulesets/Roslyn.ruleset

```diff
@@ -144,4 +144,9 @@
     <Rule Id="IDE1006" Action="Error" />                      <!-- Naming styles                                Task Open()                                         Task OpenAsync()                      -->
     <Rule Id="IDE1006WithoutSuggestion" Action="Info" />
   </Rules>
+  <Rules AnalyzerId="Microsoft.VisualStudio.Threading.Analyzers" RuleNamespace="Microsoft.VisualStudio.Threading.Analyzers">
+    <Rule Id="VSTHRD200" Action="None" />                       <!-- Naming styles                              Task Open()                                         Task OpenAsync()                      -->
+    <Rule Id="VSTHRD010" Action="None" />                       <!-- Visual Studio service should be used on main thread explicitly.-->
+    <Rule Id="VSTHRD103" Action="None" />                       <!-- Call async methods when in an async method.-->
```

Comment:
The only place this fired was here: [CreateOrUpdateNodeAsync](https://github.com/dotnet/project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/GroupedByTargetTreeViewProvider.cs#L336).  For better or worse `CreateOrUpdateNodeAsync` is not a replacement for `CreateOrUpdateNode`.  In fact `CreateOrUpdateNodeAsync` calls `CreateOrUpdateNode`.  I can change the names so the analyzer doesn't make this assumption about these private methods, but this seems fundamentally noisy to assume similarly named methods have identical functionality.